lib/linguist/heuristics.yml

```diff
@@ -383,13 +383,17 @@ disambiguations:
   rules:
    # Postgres
   - language: PLpgSQL
-    pattern: '(?i:^\\i\b|AS \$\$|LANGUAGE ''?plpgsql''?|SECURITY (DEFINER|INVOKER)|BEGIN( WORK| TRANSACTION)?;)'
+    pattern: '(?i:^\\i\b|AS \$\$|LANGUAGE ''?plpgsql''?|SECURITY (DEFINER|INVOKER)|BEGIN( WORK )?;)'
   # IBM db2
   - language: SQLPL
     pattern: "(?i:(alter module)|(language sql)|(begin( NOT)+ atomic)|signal SQLSTATE '[0-9]+')"
   # Oracle
   - language: PLSQL
     pattern: '(?i:\$\$PLSQL_|XMLTYPE|sysdate|systimestamp|\.nextval|connect by|AUTHID (DEFINER|CURRENT_USER)|constructor\W+function)'
+  # T-SQL
+  - language: TSQL
```

Comment:
The problems you experienced and mentioned at https://github.com/github/linguist/issues/4476#issuecomment-479202687 are caused by the fact your `pattern` and `negative_pattern` aren't being considered together. You need to "and" them like so:

```yaml
  # T-SQL
  - language: TSQL
    and:
      - negative_pattern: '(?i:IDENTIFIED|NUMBER|VARCHAR2)'
      - pattern: '(?i:(GO)|(CREATE PROCEDURE)|OUTPUT( INSERTED)|IF|ELSE|CURSOR|FETCH|DEALLOCATE|DECLARE)'
```

This will solve the problem you've encountered before, but will now cause a switch of classification for the `check_reorg.sql` file from SQLPL to TSQL so you're going to need a few more tweaks.

Reply:
I will make that update and then check out what other kind of tweaks I can add for remove the incorrect classification. Thanks!